### PR TITLE
Added pyhash as a package

### PR DESCRIPTION
**Pull request by thewchan**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by thewchan** [comment]
@conda-forge/help-c-cpp @conda-forge/help-python-c Might need some help with win build here, seems like it got confused with some path situation?

**Comment by thewchan** [comment]
@conda-forge/staged-recipes Calling for help again re: win build.

File: recipes/pyhash/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - cpuid
```
Comment (carterbox) [inline recipes/pyhash/meta.yaml:21]:
```suggestion
    - cpuid  # [x86]
```
https://github.com/flier/pyfasthash/blob/20a53f9bb7bf15f98e3e549f523b49e1e0f62e15/setup.py#L272

File: recipes/pyhash/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    - pip
+    - pybind11 >=2.8
+    - pytest-benchmark
+    - pytest-runner
```
Comment (carterbox) [inline recipes/pyhash/meta.yaml:25]:
```suggestion
```
None of these are required at runtime. They are for tests only.

File: recipes/pyhash/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pyhash-{{ version }}.tar.gz
+  sha256: cff5c81d613163fc59d623d4546d9be55b46ecd0e573b59057b1bb112a497763
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
```
Comment (carterbox) [inline recipes/pyhash/meta.yaml:14]:
```suggestion
build:
  script: {{ PYTHON }} -m pip install . -vv
  number: 0
  script_env:
   - PYTHON_HOME=$PREFIX
```
https://github.com/flier/pyfasthash/blob/20a53f9bb7bf15f98e3e549f523b49e1e0f62e15/setup.py#L75-L94

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/pyhash/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - cpuid  # [x86]
+    - pip
+    - pybind11 >=2.8
+    - python
+    - setuptools >=42,<58
```
Comment (carterbox) [inline recipes/pyhash/meta.yaml:27]:
```suggestion
    - setuptools >=42,<58
    # Setup checks for pytest at build time
    - pytest-benchmark
    - pytest-runner
```

**Comment by carterbox** [comment]
Now the errors are compiler syntax errors. You'll have to ask upstream about those.

**Comment by thewchan** [comment]
@conda-forge/staged-recipes @carterbox I checked upstream and it seems like they do not commit to support win, so I just skipped it. This is ready for final review then.

**Pull request merged**